### ISSUE TITLE
Impact of Copying/Recategorization

### ISSUE THREAD

**Issue by sheng-z** [ChunchuanLv/AMR_AS_GRAPH_PREDICTION#5]
Hi Lyu, impressive work!
I found that a large portion of the preprocessing code is to build the copying dictionary and recategorization, which seems nontrivial. Since you didn't report the ablation study for this part in the paper, I wonder

1. How will the model perform w/o copying/recategorization?
2. Suppose a parser takes (preprocessed-and-linearized) AMR strings and (preprocessed) English strings as input, is it possible to make use of your preprocessing step in that parser? because it's more fair to compare based on the same preprocessed input, right?

Thanks!

**Comment by ChunchuanLv** [comment]
1. As large vocabulary might pose some training challenge (so copy is
needed), and recategorizing is needed to make one to one alignment, I don't
think it is feasible to perform ablation here.

2. If you replace sequence tagging problem with latent alignment by
seq2seq, it seems every thing eles could be kept as the same. However, I am
not sure how relation prediction should interact with recategorization
system.

Regards,
Chunchuan

**Comment by jcyk** [comment]
Hi Chunchuan, 
The preprocessing seems very complicated.
I wonder if a simple way that makes the transformation between a raw AMR and its preprocessed AMR exists.
i mean the input and output are raw AMR text.

Thanks!

**Comment by sheng-z** [comment]
> Hi Chunchuan,
> The preprocessing seems very complicated.
> I wonder if a simple way that makes the transformation between a raw AMR and its preprocessed AMR exists.
> i mean the input and output are raw AMR text. And that should solve @sheng-z ‘s question.
> 
> Thanks!

Yes. I've read some AMR papers recently. Nearly all of them do recategorization in preprocessing, but few give details about it. Everyone seems to reinvent the wheel (at least partially).
Recategorization now seems a necessary step in AMR parsing. Sadly afaik no one has yet conducted a throughout analysis on it. This could make people upset since it's unclear how much improvement is from the recategorization.

@ChunchuanLv, it would be a great contribution if anyone can standardize or just detail the following pre- and post-processing steps so research can be really focused on parsing rather recategorization, and thus parsing algorithms comparison can be made more fairly.

```
# Preprocessing
AMR -> [Recategorize] -> recategorized AMR

# Training/Testing
txt -> [Parser] -> recategorized AMR

# Postprocessing
recategorized AMR -> [Recover] -> AMR
```

Best,
Sheng

**Comment by ChunchuanLv** [comment]
Indeed this is a problem, we actually think the amr treebank people should
provide such conversion (a lexicon).  There are actually resources like
https://amr.isi.edu/doc/amr-dict.html , but it is only human readable.
Also, the conversion might be hard to standardise, because your algorithm
might want to handle things differently. For example, seq2seq only want to
linearize AMR, but keep relations and node together, while my model
actually goes like this:

# Training/Testing
txt -> [Parser] -> recategorized AMR

# Postprocessing
recategorized AMR -> [Recover] -> AMR concepts -> [relation prediction] -> AMR


But in principle if such a scheme exist it could be a good starting point.

Chunchuan

**Comment by jcyk** [comment]
@ChunchuanLv thanks for your thoughts. 
Just another view of your model:
So different from others, the heavy pre-/post-processing happens only around the concept identification stage as your design. The data flow can be draw as:
txt -> recategorized concepts -> concepts
Then what we are looking for (or to study its impact) is:
#preprocessing 
AMR -> recategorized concepts
#postprocessing
recategorized concepts -> concepts

If a simple implementations of the above two transformations exist, that will help solve the questions.
